shaders.go

```diff
@@ -9,7 +12,7 @@ const bufferSize = 10000
 type Shader interface {
 	Initialize(width, height float32)
 	Pre()
-	Draw(texture *webgl.Texture, buffer *webgl.Buffer, x, y, rotation float32)
+	Draw(texture *webgl.Texture, buffer *webgl.Buffer, width, height, x, y float32, rotation float64)
```

Comment:
Why not use `float32` for this? (as we're using `float32` everywhere else)

(I know native `float64` is faster in Go, but conversion between them is even worse; see tests at https://github.com/luxengine/math)